pulsar-client/src/main/java/org/apache/pulsar/client/impl/PulsarClientImpl.java

```diff
@@ -737,4 +720,36 @@ private static Mode convertRegexSubscriptionMode(RegexSubscriptionMode regexSubs
             return null;
         }
     }
+
+    private SchemaInfoProvider loadSchemaProvider(String topicName) {
```

Comment:
nit: since there is no action of "loading", it might be better to call it `newSchemaProvider`?

```suggestion
    private SchemaInfoProvider newSchemaProvider(String topicName) {
```